### PR TITLE
Remove all ifdefs related to older versions of Qt.

### PR DESCRIPTION
**Pull request by wodim**
Now that we only support Qt 5 we will no longer need these.

### CODE REVIEW

**Comment by staticfox** [comment]
May as well grab https://github.com/wodim/KVIrc/blob/qt5/src/kvirc/ui/KviInputEditor.cpp#L1775 as well

**Comment by wodim** [comment]
I left it because of: https://github.com/wodim/KVIrc/blob/qt5/src/kvirc/ui/KviInputEditor.cpp#L1137-L1141

I also left the IPC stuff in place because god knows what it does (at least we are certain it's broken)

**Comment by un1versal** [comment]
+1 for cleanups 

> I also left the IPC stuff in place because god knows what it does (at least we are certain it's broken)

IPC is broken in windows at least there was https://github.com/kvirc/KVIrc/pull/1892 but theres still no fix for it.
Also you know what it does, its already been explained to you in the above closed PR.

File: src/kvilib/core/KviQString.cpp [hunk 1]
```diff
@@ -38,10 +38,6 @@
 #include <stdio.h> // for sprintf()
 #include <QRegExp>
 
-#if (QT_VERSION < 0x050000)
-#include <QTextDocument> // for Qt::escape()
```
Comment (un1versal) [inline src/kvilib/core/KviQString.cpp:42]: ~~Why not adding the `#include <QTextDocument>`?~~

nm I see reason now.

File: src/kvirc/ui/KviInputEditor.cpp [hunk 2]
```diff
@@ -63,16 +63,11 @@
 #include <QFontMetrics>
 #include <QKeyEvent>
 #include <QDragEnterEvent>
-#if (QT_VERSION < 0x050000)
-	#include <QInputContext>
```
Comment (un1versal) [inline src/kvirc/ui/KviInputEditor.cpp:67]: why not add `#include <QInputContext>`? and also what about the https://github.com/wodim/KVIrc/blob/qt5/src/kvirc/ui/KviInputEditor.cpp#L1136-L1152 why leave the else and why not remove the `#if (QT_VERSION >= 0x050000)`

**Comment by un1versal** [comment]
I was trying to understand some of these changes, can anyone explain and reply my questions?

**Comment by wodim** [comment]
Told ya, I left it there because the Qt 5 helper functions are still not complete.

**Comment by un1versal** [comment]
told me where, cant see a reply related to that.

> I left it because of: https://github.com/wodim/KVIrc/blob/qt5/src/kvirc/ui/KviInputEditor.cpp#L1137-L1141

that doesnt explain why you left the else for other qt version anyway

**Comment by wodim** [comment]
https://github.com/kvirc/KVIrc/pull/1924#issuecomment-206819273

File: src/kvirc/ui/KviInputEditor.cpp (hunk 2 above)
Comment (ghost) [inline src/kvirc/ui/KviInputEditor.cpp:67]: `QInputContext` was removed from Qt in Qt5, so we need this version check.

**Comment by un1versal** [comment]
idk its probably me, but the https://github.com/wodim/KVIrc/blob/qt5/src/kvirc/ui/KviInputEditor.cpp#L1142-L1148 which you left will never be called for Qt5 so again, why leave it?

**Comment by wodim** [comment]
I left it there because there's no equivalent of that code for Qt 5 for now, but in the future there probably will. So it's sort of a TODO. Read the comment above the code.

**Comment by un1versal** [comment]
yes but that code isnt run, or is it run? I did read comment, I still dont understand it and you dont care to explain clearly, if the code I pointed to in https://github.com/wodim/KVIrc/blob/qt5/src/kvirc/ui/KviInputEditor.cpp#L1142-L1148 is run then youre correct else makes no sense.

**Comment by ghost** [comment]
The code isn't run in Qt5, but since Qt5 doesn't have a solution for this, think of that block as a comment block for anyone that wants to update or check if newer Qt5/Qt6 versions have it built-in. Removing the code would hide a potential problem.

**Comment by un1versal** [comment]
as a comment yes thats obvious helpful, as is just left makes no sense, which is why I asked trying to understand reasoning.

**Comment by wodim** [comment]
It's not compiled so what is the difference between that and a comment? It's the same as comenting something by wrapping it inside `#if 0` tags.

**Comment by un1versal** [comment]
the difference is some code sanity, it seems loads of stuff is just left in code waiting for years for some unknown reason that soon gets forgotten about.

if anyone wanted to make code cleanups its utterly messy.

**Comment by Arfrever** [comment]
Qt 5 (at least 5.5.1) has `QPlatformInputContext`. Maybe it could be used?

**Comment by un1versal** [comment]
Qt 5.5 isnt default in all distros, you be OK for windows, and I doubt anyone is going to package and compile Qt in all these Other OS/distros flavours specially..

**Comment by Arfrever** [comment]
I have now checked and `QPlatformInputContext` class is present in Qt 5.0.0:
https://code.qt.io/cgit/qt/qtbase.git/tree/src/gui/kernel/qplatforminputcontext.h?h=v5.0.0

**Pull request merged**